### PR TITLE
Add button to duplicate views

### LINKED ISSUE #477: Copy view (from list and single-view page)
## Description
A user should be able to copy a view so as to not have to start from scratch when doing recurring tasks that require new views (e.g. organizing the same thing, the same way, every year).

## Relevant Job Stories
* When I'm about to create a new view, I want to be able to use an existing one as a template or copy an existing one, so that I don't have to start from scratch

## Prerequisites
No prerequisites, although the ellipsis menu referenced in this and multiple other issues does not yet exist at the time of writing.

## Requirements
- [x] There should be an ellipsis menu everywhere that a view is represented (suggested card, list, and single-view page)
- [ ] The ellipsis menu should contain an option to "Duplicate view"
- [ ] Clicking to duplicate view should create a new view with the same columns, content query and rows (if any) as the original one, and name it 'Copy of "Original name"', then navigate to the new view

## Possible implementations
Look at how creating a view is implemented using a "middle-end" API endpoint. Something similar should be possible to do for copying a view, so that the NEXT back-end can make all the necessary Zetkin API requests without having to pass data to/from the client.

## Design specifications (old wireframes but still applicable)
image

image

## Open questions
* Is the "copy columns/everything" option (nice-to-have) good, or should we have two separate options in the menu?

### PR DESCRIPTION
**Pull request by Herover**
## Description
This PR adds a button on people page in the ellipsis menus for duplicating lists.

## Screenshots
image

## Changes
* Changes the ellipsis menu on lists/views which now also shows a duplicate button if the selected item is a list, and now a folder.

## Notes to reviewer

Create a list with manually added people, and a "smart list". Both should be fully duplicated.

* Should we refer to this as duplicate or copy?
* Is it fine to use the title from the old list + " - duplicate"? (see screenshot)

## Related issues
Resolves #477

### CODE REVIEW

File: src/features/views/rpc/copy/server.ts [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+  handler: handle,
+  name: 'copyView',
+  schema: paramsSchema,
+};
+
+type ZetkinViewPostBody = Pick<ZetkinView, 'title'> & {
+  folder_id?: number;
+};
+
+async function handle(params: Params, apiClient: IApiClient): Promise<Result> {
```
Comment (richardolsson) [inline src/features/views/rpc/copy/server.ts:21]: Great solution, this! 💯

File: src/features/views/l10n/messageIds.ts [hunk 2]
```diff
@@ -30,6 +30,8 @@ export default makeMessages('feat.views', {
     },
     menu: {
       delete: m('Delete'),
+      duplicate: m('Duplicate'),
+      folderCopy: m<{ folderName: string }>('{folderName} - copy'),
```
Comment (richardolsson) [inline src/features/views/l10n/messageIds.ts:34]: I think this way of naming the copied view is good, i.e. the language of adding `- copy`. But the name of the message, `folderCopy` and the name of the parameter `folderName`, feel odd to me, because this is not related to folders.

**Review by richardolsson — CHANGES_REQUESTED** [review]
Awesome work! I'm super impressed with how quickly and nicely you solved this. It's more or less ready to merge IMO. I did just have one minor comment about naming below.

File: src/features/views/l10n/messageIds.ts (hunk 2 above)
Comment (Herover) [reply src/features/views/l10n/messageIds.ts]: Updated it now :)

**Comment by richardolsson** [comment]
Nice work @Herover! As you may have noticed, there was a minor linting issue. I pushed a commit to fix it, because I would really love to include your work in our next release, today.

**Review by richardolsson — APPROVED** [review]
This seems to be ready to merge! Nice work on all of this! 💯
